sw/device/lib/dif/dif_sram_ctrl.h

```diff
@@ -11,256 +11,276 @@
+   * been reset to the default values and all subsequent memory requests will
+   * be blocked. This condition is terminal.
+   */
+  kDifSramCtrlStatusEscalated = (0x1 << 3),
+  /**
+   * New scrambling key has been successfully obtained from OTP. If the flag is
+   * not set, the SRAM contents are still scrambled, but a default all-zero key
+   * and nonce are used to do so.
+   */
+  kDifSramCtrlStatusScrKeyValid = (0x1 << 4),
+  /**
+   * Scrambling key has been derived from a valid key seed in OTP. When
+   * `kDifSramCtrlStatusScrKeyValid` is set, but this flag is unset - the
+   * scrambling key is still ephemeral (i.e., it is derived using entropy from
+   * CSRNG), but a default all-zero value is used as the key seed. This could
+   * happen when the scrambling key seeds have not yet been provisioned to OTP.
+   */
+  kDifSramCtrlStatusScrKeySeedValid = (0x1 << 5),
+  /**
+   * Hardware initialization triggered via `dif_sram_ctrl_scramble` or
+   * `dif_sram_ctrl_wipe` has completed.
+   */
+  kDifSramCtrlStatusInitDone,
+} dif_sram_ctrl_status_t;
 
 /**
- * The result of a SRAM Controller operation.
+ * SRAM Controller lockable functionality.
  */
-typedef enum dif_sram_ctrl_result {
-  /**
-   * Indicates that the operation succeeded.
-   */
-  kDifSramCtrlOk = 0,
+typedef enum dif_sram_ctrl_lock {
   /**
-   * Indicates some unspecified failure.
+   * SRAM scrambling key renewal and "wiping" lock, which includes the following
+   * API: `dif_sram_ctrl_scramble`, `dif_sram_ctrl_request_new_key`
+   * and `dif_sram_ctrl_wipe`.
    */
-  kDifSramCtrlError,
+  kDifSramCtrlLockCtrl = 0,
   /**
-   * Indicates that some parameter passed into a function failed a
-   * precondition.
+   * Code execution from SRAM lock, which includes the following API:
+   * `dif_sram_ctrl_exec_set_enabled`.
    *
-   * When this value is returned, no hardware operations occurred.
+   * Note: this setting may not be available depending on the OTP configuration
+   * of the chip (EN_SRAM_IFETCH fuse).
    */
-  kDifSramCtrlBadArg,
-} dif_sram_ctrl_result_t;
+  kDifSramCtrlLockExec,
+} dif_sram_ctrl_lock_t;
 
 /**
- * A toggle state: enabled, or disabled.
+ * SRAM Controller DIF result codes.
+ *
+ * These codes represent an extension to the common `dif_result_t`, and are
+ * intended to be used in the DIF operations that may result in these extra
+ * result codes. For the rest, the common `dif_result_t` should be used.
  *
- * This enum may be used instead of a `bool` when describing an enabled/disabled
- * state.
+ * Due to these extended error codes being SRAM Controller DIF wide, and not
+ * on per-function basis, extra care is required by the DIF implementer, to
+ * ensure that a meaningful error code for a particular operation is returned.
  */
-typedef enum dif_sram_ctrl_toggle {
-  /**
-   * The "enabled" state.
-   */
-  kDifSramCtrlToggleEnabled,
+typedef enum dif_sram_ctrl_result {
```

Comment:
You can remove this enum entirely. We use global DIF return codes now. See `sw/device/lib/dif/dif_base.h`.